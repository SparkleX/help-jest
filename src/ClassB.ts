export class ClassB {
    public toBeMocked(a:string): string {
        console.debug("ERROR: ClassB.toBeMocked invoked")
        return "100";
    }
}

const oClassB = new ClassB ();
export { oClassB }