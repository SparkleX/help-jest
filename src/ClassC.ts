export class ClassC {
	static instance = new ClassC();
	public static get Current(): ClassC {
		return ClassC.instance;
	}
	public hello(): void{
        console.debug("hello");
    }
}