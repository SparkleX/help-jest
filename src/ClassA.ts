import { oClassB } from "./ClassB";
import { ClassC } from "./ClassC";

export class ClassA {
	public testStaticInstance(): void {
		ClassC.Current.hello();
	}
    public toBeTestA(): number {
        return 100;
    }
    public toBeTestB(): string {
        return oClassB.toBeMocked("123");
    }
}