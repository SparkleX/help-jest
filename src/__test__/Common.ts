import { ClassC } from "../ClassC";

const mockHello = jest.fn();
const oClassC: ClassC = {
	hello : mockHello
};

jest.spyOn(ClassC, 'Current', 'get').mockReturnValue(oClassC);

export { mockHello };