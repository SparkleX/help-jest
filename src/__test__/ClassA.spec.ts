import { oClassB } from "../ClassB";
import { ClassA } from "../ClassA";
import { ClassC } from "../ClassC";
import { mockHello } from "./Common";

const mock = jest.fn();
oClassB.toBeMocked = mock;



test("calla",  () => {
    const a: ClassA = new ClassA();
    const rt = a.toBeTestA();
    expect(rt).toStrictEqual(100);
});

test("mock",  () => {
	mock.mockReturnValueOnce("100");	
    const a: ClassA = new ClassA();
	const rt = a.toBeTestB();
    expect(rt).toStrictEqual("100");
	//expect(mock.mock.calls.length).toBe(1);
	//expect(mock.mock.calls[0][0]).toBe("123");
	//expect(mock.mock.calls).toContainEqual(["123","p2","p3"]);
	expect(mock.mock.calls).toMatchSnapshot("check mock calls");
});

test("static instance",  () => {
	const a: ClassA = new ClassA();
	a.testStaticInstance();
	expect(mockHello.mock.calls.length).toBe(1);
});