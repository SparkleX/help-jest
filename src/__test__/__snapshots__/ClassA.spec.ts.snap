// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock: check mock calls 1`] = `
Array [
  Array [
    "123",
  ],
]
`;
